src/main/java/com/googleinterns/zoomtube/servlets/IconFeedbackServlet.java

```diff
@@ -62,6 +68,40 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     datastore.put(createEntityFromRequest(request));
   }
 
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Optional<String> getRequestError = validateGetRequest(request);
+    if (getRequestError.isPresent()) {
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST, getRequestError.get());
+      return;
+    }
+
+    List<IconFeedback> lectures = getIconFeedback(request);
+    Gson gson = new Gson();
+    response.setContentType("application/json");
+    response.getWriter().println(gson.toJson(lectures));
+  }
+
+  /** Returns IconFeedback (associated with a specific lecture) from the database. */
+  private List<IconFeedback> getIconFeedback(HttpServletRequest request) {
```

Comment:
Just pass in the lectureId given that is what getIcconFeedback needs. You should never pass in more than what a method needs.

Reply:
fixed